reo/validators.py

```diff
@@ -1053,6 +1085,14 @@ def check_special_cases(self, object_name_path, template_values=None, real_value
                             if v is None:
                                 self.input_data_errors.append('CHP is missing a value for the {} parameter'.format(k))
 
+                        if real_values.get("chp_unavailability_hourly") is None:
+                            self.input_data_errors.append('Must input an 8760 profile for CHP unavailability')
```

Comment:
When does this case pop-up? (I thought that we are providing default values for `chp_unavailability_hourly`?)

Reply:
This is the case where the user does **not** input a `prime_mover`, and in this case, the user must provide the `chp_unavailability_hourly` input as they must for all other cost and performance parameters because we don't have defaults if they don't specify `prime_mover`. We are effectively making `prime_mover` a required input for CHP (if the user doesn't want to input all parameters) but we wanted to allow a completely custom CHP option in which all input parameters are required.

Reply:
As we discussed offline, currently `prime_mover` is required, and is limited to the `restrict_to` list in nested_inputs.py.  Do you want to keep that behavior and remove the special case handling in the validator, or allow users to not provide the `prime_mover` input? If so you might be able to leverage the `replacement_sets` option in nested_inputs.py.

Reply:
@NLaws thanks for the reminder. I updated the way `CHP.prime_mover` is validated in [this commit to chpbeta](https://github.com/NREL/REopt_Lite_API/commit/8772e8db089c62d89dc80119c0d60df06be975b5) to preserve this case where `CHP.prime_mover` is **not input** while enforcing that all other cost and performance parameters are input.